Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp

```diff
@@ -265,13 +271,15 @@ void Recorder::drawPattern(ImageBuffer& imageBuffer, const FloatRect& destRect,
 
 void Recorder::save()
 {
+    appendStateChangeItemIfNecessary();
     GraphicsContext::save();
     recordSave();
     m_stateStack.append(m_stateStack.last());
 }
 
 void Recorder::restore()
 {
+    appendStateChangeItemIfNecessary();
```

Comment:
Nope, you're pushing for introducing the exact same bug. :)
It's also tested in the tests:
```
c.save();
c.setShouldAntialias(false);
c.restore();
c.fillRect({..});
```

Here, if you don't enforce that the state is committed before `restore()`, this will happen:
```
c.save();
c.restore();
c.setShouldAntialias(false);
c.fillRect({..});
```
(The fillRect is the first one that would then commit the state)